gridfs/synchronous/grid_file.py

```diff
@@ -1015,6 +1015,36 @@ def rename(
                 "matched file_id %i" % (new_filename, file_id)
             )
 
+    def rename_by_name(
```

Comment:
Let's add some hand written tests for this too like test_rename in test_gridfs_bucket. That way we get some signal on typing and the UX.